packages/sdk/src/manifold/types.ts

```diff
@@ -0,0 +1,81 @@
+export enum ListingType {
+  INVALID,
+  INDIVIDUAL_AUCTION,
+  FIXED_PRICE,
+  DYNAMIC_PRICE,
+  RANKED_AUCTION,
+}
+
+export enum Spec {
+  NONE,
+  ERC721,
+  ERC1155,
+}
+
+export type BNHex = {
+  type: string;
+  hex: string;
+};
+
+type BaseListing = {
+  id: string;
+  seller: string;
+  fees: DeliveryFees;
+};
+
+export type DeliveryFees = {
+  deliverFixed: number | null;
+  deliverBPS?: number;
+};
+
+// FROM API
+export type ApiListing = BaseListing & {
+  details: ApiListingDetails;
+  token: ApiListingTokenDetails;
+};
+
+// FOR CONTRACT
+export type ContractListing = BaseListing & {
+  details: ListingDetails;
+  token: ListingTokenDetails;
+};
+
+type BaseDetails = {
+  type_: ListingType;
+  extensionInterval: number;
+  startTime: number;
+  endTime: number;
+};
+
+export type ApiListingDetails = BaseDetails & {
+  initialAmount: BNHex;
+  totalAvailable: string;
+  totalPerSale: string;
+  minIncrementBPS?: string;
+  erc20: string | null;
+  identityVerifier: string | null;
+};
+
+export type ListingDetails = BaseDetails & {
+  initialAmount: string;
+  totalAvailable: number;
+  totalPerSale: number;
+  minIncrementBPS: number;
+  erc20: string;
+  identityVerifier: string;
+};
+
+type BaseTokenDetails = {
+  address_: string;
+  lazy: boolean;
+};
+
+export type ListingTokenDetails = BaseTokenDetails & {
+  id: string;
+  spec: Spec;
+};
+
+export type ApiListingTokenDetails = BaseTokenDetails & {
```

Comment:
We want to be fully unaware of the existence of the API, we just want a standard order format that the API response can be converted to. We should probably remove any API-specific types/fields.

Reply:
These seems to be the all the details of the order that are available on-chain (and that we want to save):
```
    event CreateListing(uint40 indexed listingId, uint16 marketplaceBPS, uint16 referrerBPS, uint8 listingType, uint24 totalAvailable, uint24 totalPerSale, uint48 startTime, uint48 endTime, uint256 initialAmount, uint16 extensionInterval, uint16 minIncrementBPS, address erc20, address identityVerifier);
    event CreateListingTokenDetails(uint40 indexed listingId, uint256 id, address address_, uint8 spec, bool lazy);
    event CreateListingFees(uint40 indexed listingId, uint16 deliverBPS, uint240 deliverFixed);
```